### PR TITLE
Add constructor for mpl3115a2

### PR DESCRIPTION
**Pull request by halaney**
Added a constructor for mpl3115a2 that attempts to open the i2c adapter
given and confirm via and i2c read that the device located at the
address is a mpl3115a2.

### CODE REVIEW

File: mpl3115a2.cpp [hunk 1]
```diff
@@ -1,9 +1,65 @@
...
+    {
+        std::ostringstream err;
+        err << "Could not set slave address\n" << strerror(errno);
+        throw std::runtime_error(err.str());
+    }
+
+    // Confirm that the device at this address is indeed the MPL3115A2
+    uint8_t buf[1];
+    buf[0] = WHO_AM_I;
+    if (read(m_i2cFile, buf, 1) != 1)
```
Comment (halaney) [inline mpl3115a2.cpp:47]: should prolly factor this out later or make a i2c class and inherit

File: mpl3115a2.cpp [hunk 2]
```diff
@@ -1,9 +1,65 @@
...
+    {
+        std::ostringstream err;
+        err << "WHO_AM_I register contained: " << std::hex
+            << static_cast<int>(buf[0]) << std::endl
+            << "Expected: " << std::hex << 0xC4 << std::endl;
+        throw std::runtime_error(err.str());
+    }
+    else
+    {
+        std::cout << "MPL3115A2 confirmed to on I2C bus represented by "
```
Comment (samahart) [inline mpl3115a2.cpp]: confirmed to _be_ on I2C

File: mpl3115a2.cpp [hunk 3]
```diff
@@ -1,9 +1,65 @@
...
+    // Confirm that the device at this address is indeed the MPL3115A2
+    uint8_t buf[1];
+    buf[0] = WHO_AM_I;
+    if (read(m_i2cFile, buf, 1) != 1)
+    {
+        std::ostringstream err;
+        err << "Could not read WHO_AM_I register" << strerror(errno);
+        throw std::runtime_error(err.str());
+    }
+    if (buf[0] != 0xC4)
```
Comment (samahart) [inline mpl3115a2.cpp]: What does 0xC4 represent?

**Review by samahart — APPROVED** [review]
Looks good. I like it.

**Pull request merged**